testdata/testoutput5

```diff
@@ -1910,6 +1910,24 @@ No match
         End
 ------------------------------------------------------------------
 
+/\p{Any}#\P{Any}#[\p{Any}]#[\P{Any}]#[\p{Any}a-z]#[\P{Any}c]/B,utf
+------------------------------------------------------------------
+        Bra
+        AllAny
+        #
+        []
+        #
+        AllAny
+        #
+        []
+        #
+        AllAny
+        #
+        [c]
```

Comment:
This is not caught by the `[c]` -> `c` optimization, but I think it is not worth to make it more complex.

Reply:
That seems fair. I also deliberately skipped those optimisations too for things like `(?[ [ab] - [b] ])` because I felt it was just too obscure.